libs/constrained-generators/src/Constrained/GenT.hs

```diff
@@ -170,6 +170,16 @@ listOfT gen = do
   lst <- pureGen . listOf $ runGenT gen Loose
   catGEs lst
 
+-- TODO: possibly one could return "more, fewer, ok" in the `validLen` instead
+-- of `Bool`
+listOfUntilLenT :: MonadGenError m => GenT GE a -> Int -> (Int -> Bool) -> GenT m [a]
```

Comment:
please add a comment, that explains what this does. Hard to understand, what we could possibly add, if we don't know what it currently does.